analyzer/src/main/kotlin/PackageManager.kt

```diff
@@ -301,3 +302,15 @@ abstract class PackageManager(
  */
 fun parseAuthorString(author: String?, vararg delimiters: Char = charArrayOf('<')): String? =
     author?.split(*delimiters, limit = 2)?.firstOrNull()?.trim()?.ifEmpty { null }
+
+private fun PackageManagerResult.addDependencyGraphIfMissing(): PackageManagerResult {
```

Comment:
Adding an empty `DependencyGraph` should fix the problem; however, I am not sure whether there is a more direct solution. The check in `CompatibilityDependencyNavigator` whether a project uses the tree format could be changed as well. Would it be sufficient to change it to something like

`private fun Project.usesTree(): Boolean = scopeNames?.isEmpty() ?: true` ?

So, the dependency graph format is only assumed if there are actually scope names available. This change does not break the unit tests; they are still green.

Reply:
> Adding an empty `DependencyGraph` should fix the problem;

I've verified that it does for the case where it happened.

> Would it be sufficient to change it to something like `private fun Project.usesTree(): Boolean = scopeNames?.isEmpty() ?: true ?`

IIUC it would, but I haven't testet it. I thought about this option as well and decided against it because

1. The function name wouldn't match anymore the return value
2. I didn't want to migrate the case from `Graph` to `Tree` because I assumed that our plan is to get rid of the `Tree` in the future. So I thought it moves into the wrong direction.

Anyhow, I can also try changing it if you still think it makes sense. @oheger-bosch do you want me to?

Reply:
I am a bit on the fence. IIUC, this is about a corner case, in which there are neither scope names nor scopes (in the old format). I would agree that the graph format is the new one and should be used as far as possible; so this is an argument for the proposed solution.

There are, however, two things I do not like with this approach:
* The fix is done at a different place. I mean, it is not directly obvious that a change in `PackageManager` is necessary to impact the selection of the dependency navigator.
* To determine whether a dummy dependency graph needs to be created, logic is executed that is slightly different from the one contained in the `usesTree()` function. Ideally, `PackageManager` should not know about these details at all.

Well, these are no strong objections, just a gut feeling. So I leave the decision, which way to follow, to you.

Reply:
I am following you with having things in two places PackageManager and CompatibilityNavigator is not nice. 
However, the package manager currently does construct the OrtResult and I don't see a way to eliminate the construction from the package manager. On the other hand the navigator has some precondition checks. So the only flexibility / alternative I can see is to change the require condition [1] to tolerate the absence of the graph which implies the need for adding a graceful handling. 

So in total we have 3 options:
1. implementation as-is
2. change `usesTree()`
3. change the require condition [1] and add handling

My preference is still towards option 1, because I cannot see clearly much benefit in using the other opions.
Anyhow, If you want me to change it I will do it.

[1] https://github.com/oss-review-toolkit/ort/blob/463a6ebb57cd3e9613b954a2094a509947c85dc3/model/src/main/kotlin/DependencyGraphNavigator.kt#L29-L31

Reply:
Option 3 would also be fine with me. For option 1 I see the drawback that an `OrtResult` can be constructed manually, for which no `DependencyNavigator` can be requested. So options 2 and 3 would be safer in this regard.